src/components/FilterBar/FilterDisplay/Inputs/DateInput.react.js

```diff
@@ -0,0 +1,73 @@
+export class DateInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { value: this.props.filterBarActor.getFilter(this.props.filterUid).value || { from: null, to: null } };
+  }
+
+  _onChange(event) {
+    var newValue = this.state.value;
+
+    if(event.type === 'dp') {
+      newValue[event.target.querySelector('input').getAttribute('placeholder')] = event.target.querySelector('input').value;
+    } else if (event.type === 'input') {
+      newValue[event.target.getAttribute('placeholder')] = event.target.value;
+    }
+
+    this.setState({value: newValue});
+    this.props.filterBarActor.updateFilter(this.props.filterUid, 'value', newValue);
+  }
+
+  componentDidMount() {
+    var datePickerFrom = $(React.findDOMNode(this.refs.dateRangeFrom));
```

Comment:
I want to say theres a better way to handle this. Using a jQuery selector inside react component feels like we are batting a snake with 2 sticks.. I don't know what a better way to do it would be though.